app/Repositories/PhotoRepository.php

```diff
@@ -76,6 +76,71 @@ public function create(array $data, $signupId)
     }
 
     /**
+     * Update an existing photo post and signup.
+     *
+     * @param \Rogue\Models\Photo $signup
+     * @param array $data
+     *
+     * @return \Rogue\Models\Photo
+     */
+    public function update($signup, $data)
+    {
+        // Update the signup's quantity and why_participated data.
+        // We will always update these since we can't tell if this has been changed in a good way yet.
+        $data['quantity_pending'] = $data['quantity'];
+        $signup->fill(array_only($data, ['quantity_pending', 'why_participated']));
+        $signup->save();
+
+        // If there is a file, create a new photo post.
+        if (isset($data['file'])) {
+            return $this->create($data, $signup->id);
```

Comment:
@katiecrane Just wanted to flag this for you in the code! We chatted about this briefly last week re: whether or not we need the `array_key_exists` statement similar to [this line](https://github.com/DoSomething/rogue/blob/master/app/Repositories/ReportbackRepository.php#L134) with a broken photo - let me know if we do!